### PR TITLE
Eng/pr noelle 10 8

### PR DESCRIPTION
**Pull request by Kerowyn202**

### CODE REVIEW

**Review by therawv — COMMENTED** [review]
I'm not really familiar with how you set up the paths and when I tried to execute the code it didn't run properly. I've been using relative paths from the same directory. Maybe you should adjust it?

**Comment by Kerowyn202** [comment]
Weird, it runs properly on my computer, both when I have the FileNotFound exception set up to trigger and when I don't. What error are you getting @therawv ?

**Review by therawv — CHANGES_REQUESTED** [review]
I reviewed this again today and its failing to run properly. It's failing to find the cat.txt and dog.txt files I posted a screenshot in the discord for further context.

**Comment by Kerowyn202** [comment]
Edited it to include specifications about where the program should be run in the error message. So you'll get the custom error if you 1) don't have the files or 2) don't run it from PCC and run it from any other folder

File: chapter_10/10-8.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from pathlib import Path
+
+pet_names = ['./chapter_10/cats.txt', './chapter_10/dogs.txt']
```
Comment (therawv) [inline chapter_10/10-8.py:3]: Your code would be more robust if you used a relative path to the tune of
pet_names = ['cats.txt', 'dogs.txt']

Or if you anchored the path, to the tune of what Alexa did in her chapter 10-12 (this wouldn't be the exact code but it's an example of how she roots her directory)

import json
import os
from pathlib import Path

def getRootDir():
    """ """
    # deal with "What folder am I working from?"
    cwd = os.getcwd()
    index = cwd.find("chapter")
    if index > 0:
        cwd = cwd[:index]
    # else do nothing, assume we're in workspace root
    return cwd

def main():
    """ """
    rootDir = getRootDir()

    # create scratch dir
    scratchDir = Path(f"{rootDir}/scratch_files")
    if not scratchDir.exists():
        os.mkdir(scratchDir.resolve())

This change still leaves lots of potential failure cases when a user pulls the code(I'm still getting an error where I shouldn't, and I am in the PCC directory but its not in the same base location as you). At the very least you should comment the required filepath/placement somewhere in here so a developer would know how to set it up.

**Review by therawv — COMMENTED** [review]
This change doesn't really address the cause of the code failing in the first place. There are simpler solutions to a more robust answer.

**Review by therawv — APPROVED** [review]

**Pull request merged**